CoreScriptsRoot/CoreScripts/Topbar.lua

```diff
@@ -1277,23 +1289,41 @@ local function Create3DUserGuiToggleIcon(topBarInstance, panel, menubar)
 		Image = "rbxasset://textures/ui/VR/toggle2D.png"
 	}
 
-	local userGuiPanel = Panel3D.Get("UserGui")
+	local thisModuleName = "UserGui"
+
+	local userGuiPanel = Panel3D.Get(thisModuleName)
 	userGuiPanel:SetType(Panel3D.Type.Fixed)
-	userGuiPanel:ResizePixels(300, 125)
+	userGuiPanel:ResizeStuds(4, 4, 128)
 	userGuiPanel:SetVisible(false, false)
 
-	menuItem.MouseButton1Click:connect(function()
-		if InputService.VREnabled then
-			userGuiPanel:SetVisible(not userGuiPanel.isVisible, false)
+	local VRHub = require(GuiRoot.Modules.VR.VRHub)
+
+	VRHub.ModuleOpened.Event:connect(function(moduleName, isExclusive, shouldCloseNonExclusive, shouldKeepTopbarOpen)
```

Comment:
This seems like a lot of args for this event, and it is getting kind of confusing what they all do. If VRHub is keeping state of these things perhaps we can call some functions on the VRHub to properly process this event?